src/square.cc

```diff
@@ -89,6 +90,133 @@ namespace triton { namespace backend { namespace square {
     }                                                                        \
   } while (false)
 
+//
+// ModelParameters
+//
+// Helper class for parsing and storing model config parameters, and help
+// performing required operations on the stored parameters.
+//
+class ModelParameters {
+ public:
+  enum DelayType { Infer, Output };
+  enum InferResultType { Success, Fail, Empty };
```

Comment:
nit: Shoud enum members be all capitals? INFER, OUTPUT ...

Reply:
Sure. Updated.

Reply:
No need for a change, but I just ran into issues with enums in the python backend because they were named like [macros](https://github.com/triton-inference-server/python_backend/pull/294#discussion_r1474706359). Just something to keep in mind, especially if the enum fields we're naming are common.

Reply:
Interesting... so if there is an enum
```
enum MyEnum { ERROR, SUCCESS };
```
and on some imports someone thinks it is a good idea to define a macro
```
#define ERROR 0
```
then the enum becomes
```
enum MyEnum { 0, SUCCESS };
```

Reply:
No, thankfully it won't do that. Instead, it won't compile.

For example, `windows.h` has this macro for `ERROR`:
`#define ERROR 0`

Let's say I declare the following enum in `test.cc`:
```
enum foo {
    ERROR=0,
    WARN,
    INFO,
};
```
When I try to compile it, I get the following error:
```
test.cc(10): error C2059: syntax error: 'constant'
test.cc(14): error C2143: syntax error: missing ';' before '}'
test.cc(14): error C2059: syntax error: '}'
```
Note that I get this error despite ERROR being equal to 0 in both cases.